sympy/calculus/util.py

```diff
@@ -1128,8 +1128,14 @@ def __add__(self, other):
             if other is S.Infinity and self.min is S.NegativeInfinity or \
                     other is S.NegativeInfinity and self.max is S.Infinity:
                 return AccumBounds(-oo, oo)
-            elif other.is_extended_real:
+            elif other.is_extended_real and self.min is not S.NegativeInfinity and self.max is not S.Infinity:
                 return AccumBounds(Add(self.min, other), Add(self.max, other))
+            elif self.min is S.NegativeInfinity and self.max is S.Infinity:
+                return AccumBounds(-oo, oo)
+            elif self.min is S.NegativeInfinity:
+                return AccumBounds(-oo,self.max+other)
+            elif self.max is S.Infinity:
+                return AccumBounds(self.min+other,oo)
```

Comment:
These lines should be run only when other.is_extended_real. There could be a test where other is `I`, for instance.